pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -189,13 +191,26 @@ func NewInstallerController(
 	}
 
 	c.ownerRefsFn = c.setOwnerRefs
-	c.factory = factory.New().WithInformers(operatorClient.Informer(), kubeInformersForTargetNamespace.Core().V1().Pods().Informer())
+	c.factory = factory.New().WithInformers(
+		operatorClient.Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Informer(),
```

Comment:
This is OK. I don't like that the usage of these informers is implicit.
Could you add a comment somewhere [here](https://github.com/openshift/cluster-kube-apiserver-operator/blob/ae0c9631284bca8cfcbc305674781e097f7238ae/vendor/github.com/openshift/library-go/pkg/operator/staticpod/controllers.go#L212) saying that `configMapClient` and `secretClient` use listers and the controllers must implicitly wait unit the lister will be synced.

Please also add a comment here saying that listers come from these two clients that are passed as a dependency to this controller.

Reply:
Do you mind checking if the other controllers (accepting these two clients) properly wait for synchronisation ? (Can be a follow-up PR)

Reply:
yep, let's split this PR out into two then.

Reply:
ok, thanks.

Reply:
part two here: https://github.com/openshift/library-go/pull/1743